codegen/render/manifests_renderer.go

```diff
@@ -218,41 +245,82 @@ func SetVersionForObject(obj metav1.Object, version string) {
 		}
 
 		a[crdutils.CRDVersionKey] = strippedVersion.String()
+
 		obj.SetAnnotations(a)
 	}
 }
 
-func (r ManifestsRenderer) renderManifest(appName string, mk MakeResourceFunc, group Group) (string, error) {
-	objs, err := mk(group)
-	if err != nil {
-		return "", err
+// TODO (dmitri-d): this can be removed once we migrate to use platform charts exclusively
+func (r ManifestsRenderer) renderCRDManifest(appName, groupName string, objs []apiextv1.CustomResourceDefinition) (OutFile, error) {
+	outFile := OutFile{
+		Path: r.ManifestDir + "/crds/" + groupName + "_" + "crds.yaml",
 	}
 
 	var objManifests []string
 	for _, obj := range objs {
-		// find the annotation of the manifest, and add to them
-		SetVersionForObject(obj, group.AddChartVersion)
-		manifest, err := marshalObjToYaml(appName, obj)
+		manifest, err := marshalObjToYaml(appName, &obj)
 		if err != nil {
-			return "", err
+			return OutFile{}, err
 		}
 		objManifests = append(objManifests, manifest)
 	}
 
-	return strings.Join(objManifests, "\n---\n"), nil
+	outFile.Content = strings.Join(objManifests, "\n---\n")
+	return outFile, nil
 }
 
-func marshalObjToYaml(appName string, obj metav1.Object) (string, error) {
-	labels := obj.GetLabels()
-	if labels == nil {
-		labels = map[string]string{}
+func (r ManifestsRenderer) renderTemplatedCRDManifest(appName, groupName string,
+	objs []apiextv1.CustomResourceDefinition,
+	grandfatheredGroups map[string]bool) (OutFile, error) {
+
+	renderer := DefaultTemplateRenderer
+
+	// when rendering helm charts, we need
+	// to use a custom delimiter
+	renderer.left = "[["
+	renderer.right = "]]"
+
+	defaultManifestRenderer := ChartRenderer{
+		templateRenderer: renderer,
 	}
 
-	labels["app"] = appName
-	labels["app.kubernetes.io/name"] = appName
+	outFile := OutFile{Path: r.ManifestDir + "/templates/" + groupName + "_" + "crds.yaml"}
+	templatesToRender := inputTemplates{
+		"manifests/crd.yamltmpl": outFile,
+	}
+	files, err := defaultManifestRenderer.renderCoreTemplates(
+		templatesToRender,
+		templateArgs{Crds: objs, ShouldSkip: grandfatheredGroups})
+	if err != nil {
+		return OutFile{}, err
+	}
+	return files[0], nil
```

Comment:
Files always len 1? Should we check?

Reply:
we only give `defaultManifestRenderer.renderCoreTemplates` one template to render; if there's an error it will return before this line. So, yes, if it got here, there is one item in []OutFile. I'll add a comment.